src/middlewared/middlewared/plugins/chart_releases_linux/chart_release.py

```diff
@@ -616,10 +616,12 @@ async def do_delete(self, job, release_name, options):
                 await self.middleware.call('k8s.pod.delete', pod['metadata']['name'], {'namespace': namespace})
 
         job.set_progress(75, f'Waiting for {release_name!r} pods to terminate')
+        # We still have a fallback for waiting for pods to terminate
+        # because of https://github.com/helm/helm/issues/10586
         await self.middleware.call('chart.release.wait_for_pods_to_terminate', get_namespace(release_name))
-        await self.middleware.call('chart.release.wait_for_namespace_to_terminate', get_namespace(release_name))
 
         await self.post_remove_tasks(release_name, job)
+        await self.middleware.call('chart.release.wait_for_namespace_to_terminate', get_namespace(release_name))
```

Comment:
Suggestion:
If it does not, check if there are any kubernetes objects within said namespace (including Custom Resource Objects)... If not, we can safely force nuke it.

Reply:
We are already nuking it but the API server registers the request and does not wait for the termination to complete hence the wait